install.sh

```diff
@@ -1,48 +1,51 @@
 #!/bin/bash
 
-set -Eeuo pipefail
 echo "=================================================================================";
 echo "Beginning CDash installation...";
 
-INSTALLATION_ERROR_MESSAGE="
-* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
-*                             INSTALLATION FAILURE!                             *
-*                                                                               *
-* An error occurred while upgrading your CDash installation and CDash was not   *
-* able to recover automatically.  If you believe that this is a problem with    *
-* CDash, please report the error here:                                          *
-*                                                                               *
-* -> https://github.com/Kitware/CDash/issues/new                                *
-*                                                                               *
-* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
-";
+error_handler() {
+echo "
+* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+*                            INSTALLATION FAILURE!                            *
+*                                                                             *
+* An error occurred while installing CDash and CDash was not able to recover  *
+* automatically.  If you believe that this is a problem with CDash, please    *
+* report the error here:                                                      *
+*                                                                             *
+* -> https://github.com/Kitware/CDash/issues/new                              *
+*                                                                             *
+* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+"
+
+exit 1
+}
+
+trap error_handler ERR
 
 # Temporarily change to /cdash
-{
-  pushd "/cdash" > /dev/null &&
+pushd "/cdash" > /dev/null
 
-  echo "Enabling maintenance mode..." &&
-  php artisan down --render="maintenance" --refresh=5 &&
+echo "Enabling maintenance mode..."
+php artisan down --render="maintenance" --refresh=5
 
-  echo "Updating vendor dependencies..." &&
-  npm install &&
-  composer install &&
+echo "Updating vendor dependencies..."
+npm install
+composer install
```

Comment:
The docker file already runs `npm install` and `composer install`. Running it again in the entrypoint has several effects:

- Increased startup time
- Increased resource usage -- all the package files are stored locally on the node in an overlay
- The image is no longer immutable and may introduce vulnerabilities through dependencies at runtime
- The image can no longer be ran as a read-only root with volumes mounted to the storage directory
```suggestion

```

Reply:
The docker image unfortunately has to be mutable either way because the website frontend must be built from environment variables which are only known at startup via `npm run prod`.  Running the dependency update step again should effectively be a no-op, because we pin our dependencies via lockfiles.

The intention here was to have a shared installation process which can be used for container-based systems, as well as traditional installations.  I'll do some thinking and see if I can come up with a better solution.